From stable-bounces@linux.kernel.org Tue Nov 13 00:12:09 2007
From: Patrick McHardy <kaber@trash.net>
Date: Tue, 13 Nov 2007 00:11:51 -0800 (PST)
Subject: Fix VLAN address syncing.
To: stable@kernel.org
Cc: bunk@kernel.org
Message-ID: <20071113.001151.121206681.davem@davemloft.net>

From: Patrick McHardy <kaber@trash.net>

patch d932e04a5e7b146c5f9bf517714b986a432a7594 in mainline.

[PATCH] [VLAN]: Don't synchronize addresses while the vlan device is down

While the VLAN device is down, the unicast addresses are not configured
on the underlying device, so we shouldn't attempt to sync them.

Noticed by Dmitry Butskoy <buc@odusz.so-cdu.ru>

Signed-off-by: Patrick McHardy <kaber@trash.net>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 net/8021q/vlan.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/net/8021q/vlan.c
+++ b/net/8021q/vlan.c
@@ -629,6 +629,10 @@ static int vlan_device_event(struct noti
 			if (!vlandev)
 				continue;
 
+			flgs = vlandev->flags;
+			if (!(flgs & IFF_UP))
+				continue;
+
 			vlan_sync_address(dev, vlandev);
 		}
 		break;
